tuxemon/item/effects/evolve.py

```diff
@@ -42,10 +43,15 @@ class EvolveEffect(ItemEffect):
     monster_evolve: str
 
     def apply(self, target: Monster) -> EvolveEffectResult:
-        monster_evolve = self.monster_evolve
-
-        if any(d["monster_slug"] == monster_evolve for d in target.evolutions):
-            self.user.evolve_monster(target, monster_evolve)
+        if self.monster_evolve == "random":
+            choices = [d for d in target.evolutions if d.path == d.path.item]
+            evolution = random.choice(choices).monster_slug
+            self.user.evolve_monster(target, evolution)
             return {"success": True}
+        elif self.monster_evolve != "random":
+            for evolution in target.evolutions:
+                if evolution.monster_slug == self.monster_evolve:
+                    self.user.evolve_monster(target, self.monster_evolve)
+                    return {"success": True}
         else:
             return {"success": False}
```

Comment:
Remove the final else block and deindent the final return statement. Otherwise it's theoretically possible for you to return None if line 53 is never true.

Reply:
Ok